### PR TITLE
[enhancement](fix)change ordinary type null value is \N,complex type null value is null

### PR DESCRIPTION
**Pull request by hubgeter**
## Proposed changes
1. 
For null values in ordinary types, we use \N to represent them;
for null values in nested types, we use null to represent them, just like the json format.

example:
If you have three nullable columns
a : int, b : string, c : map<string,int>
data:
```
\N,hello world,\N
1,\N,{"cmake":2,"null":11}
9,"\N",{"\N":null,null:0}
\N,"null",{null:null}
null,null,null
```
if you set trim_double_quotes = true
you will get :
```
NULL,hello world,NULL
1,NULL,{"cmake":2,"null":11}
9,\N,{"\N":NULL,NULL:0}
NULL,null,{NULL:NULL}
NULL,null,NULL
```
if you set trim_double_quotes = false
you will get :
```
NULL,hello world,NULL
1,\N,{"cmake":2,"null":11}
9,"\N",{"\N":NULL,NULL:0}
NULL,"null",{NULL:NULL}
NULL,null,NULL
```
in csv(text) for normal type: we only recognize \N for null , so
for not char family type, like int, if we put null literal ,
it will parse fail, and make result null，not just because it equals \N.
for char family type, like string, if we put null literal, it will parse success,
and "null" literal will be stored in doris.

2.
For strings in the json complex type, we remove double quotes by default.

Because when querying complex types, such as selecting complexColumn from table,
we will add double quotes to the strings in the complex type.

For the map<string,int> column, insert { "abc" : 1, "hello",2 }.
If you do not remove the double quotes, it will display {""abc"":1,""hello"": 2 },
remove the double quotes to display { "abc" : 1, "hello",2 }.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM
But please describe more in your PR comment about how to handle `null` in text

File: be/src/vec/data_types/serde/data_type_nullable_serde.cpp [hunk 1]
```diff
@@ -117,24 +119,25 @@ Status DataTypeNullableSerDe::deserialize_column_from_hive_text_vector(IColumn&
...
+                                                             const FormatOptions& options,
+                                                             int nesting_level) const {
     auto& null_column = assert_cast<ColumnNullable&>(column);
     // TODO(Amory) make null literal configurable
 
     if (!(options.converted_from_string && slice.trim_quote())) {
         //for map<string,string> type : {"abc","NULL"} , the NULL is string , instead of null values
-        if (slice.size == 4 && slice[0] == 'N' && slice[1] == 'U' && slice[2] == 'L' &&
-            slice[3] == 'L') {
+        if (nesting_level >= 2 && slice.size == 4 && slice[0] == 'n' && slice[1] == 'u' &&
```
Comment (morningman) [inline be/src/vec/data_types/serde/data_type_nullable_serde.cpp:129]: Add comment to describe the logic here. Better give some example

**Comment by hubgeter** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_nullable_serde.cpp [hunk 2]
```diff
@@ -116,25 +118,55 @@ Status DataTypeNullableSerDe::deserialize_column_from_hive_text_vector(IColumn&
...
     // sure slice is from string parse , we can parse this "null" literal as string "null" to
     // nested column , otherwise we insert null to null column
     if (!(options.converted_from_string && slice.trim_quote())) {
-        if (slice.size == 4 && Slice::mem_equal("null", slice.data, 4)) {
+        /*
+         * For null values in ordinary types, we use \N to represent them;
+         * for null values in nested types, we use null to represent them, just like the json format.
+         *
+         * example:
+         * If you have three nullable columns
```
Comment (amorynan) [inline be/src/vec/data_types/serde/data_type_nullable_serde.cpp:135]: make a note: null -> int -> NULL | null -> char family -> "null"

**Comment by hubgeter** [comment]
run buildall

**Comment by hubgeter** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**